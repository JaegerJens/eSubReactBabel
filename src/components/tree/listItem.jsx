import React from "react";

export const ListItem = ({number, value, onClick, isSelected}) => {

    const onClickHandler = () => onClick(number, value);

    if (isSelected) {

        return <li className="selected" onClick={onClickHandler}>{value}</li>;

    }

    return <li onClick={onClickHandler}>{value}</li>;

};

export default ListItem;
