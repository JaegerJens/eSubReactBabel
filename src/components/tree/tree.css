li {
    color: blue;
    text-decoration: underline;
    cursor: pointer;
}

li.selected {
    color: black;
    text-decoration: none;
}