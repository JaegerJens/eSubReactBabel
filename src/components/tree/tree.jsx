import ListItem from "./listItem.jsx";
import React from "react";

require("./tree.css");

const Tree = ({select, data, onClick}) => {

    const isSelected = (key) => key === select;
    const listData = data.map(
            (entry, index) =>
                <ListItem
                    isSelected={isSelected(index)}
                    key={index}
                    number={index}
                    onClick={onClick}
                    value={entry}
                />
    );


    return <ul>{listData}</ul>;

};

export default Tree;
