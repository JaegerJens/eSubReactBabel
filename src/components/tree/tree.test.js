import tree, {listItem} from "./tree.jsx";
import React from "React";
import renderer from "react-test-renderer";

describe("Tree component", () => {

    it("matches snapshot", () => {

        const props = {
            "data": ["a", "b", "c"],
            "onClick": "onClick function",
            "select": 2
        };

        expect(tree(props)).toMatchSnapshot();

    });

});

