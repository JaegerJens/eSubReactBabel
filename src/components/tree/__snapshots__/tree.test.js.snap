// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree component matches snapshot 1`] = `
<ul>
  <ListItem
    isSelected={false}
    number={0}
    onClick="onClick function"
    value="a"
  />
  <ListItem
    isSelected={false}
    number={1}
    onClick="onClick function"
    value="b"
  />
  <ListItem
    isSelected={true}
    number={2}
    onClick="onClick function"
    value="c"
  />
</ul>
`;
